Exercise_5.java

```diff
@@ -1,36 +1,78 @@
-class IterativeQuickSort { 
-    void swap(int arr[], int i, int j) 
-    { 
-	//Try swapping without extra variable 
-    } 
-  
-    /* This function is same in both iterative and 
+import java.util.Stack;
+
+/*
+instead of using inbuilt recursion stack, we create own stack DS to store the indices of left and right boundaries
+of sub array.
+
+ */
+class IterativeQuickSort {
+    void swap(int arr[], int i, int j)
+    {
+	//Try swapping without extra variable
+        int temp = arr[i];
+        arr[i] = arr[j];
+        arr[j] = temp;

```

Comment:
Try swapping without using extra variable

Reply:
This is interesting. Never came up or thought of we could even swap without extra space. Will do that.